tests/unittests/CoreGraphics.drawing/CGContextDrawing_FillModeTests.cpp

```diff
@@ -0,0 +1,124 @@
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .3 * height, 0, M_PI, true);
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .3 * height, M_PI, 0, true);
+
+    CGPathMoveToPoint(leftCircles, NULL, xstart + .25 * width + .2 * height, ystart + .5 * height);
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .2 * height, 0, M_PI, true);
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .2 * height, M_PI, 0, true);
+
+    CGPathMoveToPoint(leftCircles, NULL, xstart + .25 * width + .1 * height, ystart + .5 * height);
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .1 * height, 0, M_PI, true);
+    CGPathAddArc(leftCircles, NULL, xstart + .25 * width, ystart + .5 * height, .1 * height, M_PI, 0, true);
+
+    CGPathCloseSubpath(leftCircles);
+
+    CGMutablePathRef rightCircles = CGPathCreateMutable();
+
+    CGPathMoveToPoint(rightCircles, NULL, xstart + .75 * width + .4 * height, ystart + .5 * height);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .4 * height, 0, M_PI, false);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .4 * height, M_PI, 0, false);
+
+    CGPathMoveToPoint(rightCircles, NULL, xstart + .75 * width + .3 * height, ystart + .5 * height);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .3 * height, 0, M_PI, true);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .3 * height, M_PI, 0, true);
+
+    CGPathMoveToPoint(rightCircles, NULL, xstart + .75 * width + .2 * height, ystart + .5 * height);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .2 * height, 0, M_PI, false);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .2 * height, M_PI, 0, false);
+
+    CGPathMoveToPoint(rightCircles, NULL, xstart + .75 * width + .1 * height, ystart + .5 * height);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .1 * height, 0, M_PI, true);
+    CGPathAddArc(rightCircles, NULL, xstart + .75 * width, ystart + .5 * height, .1 * height, M_PI, 0, true);
+
+    CGPathCloseSubpath(rightCircles);
+
+    CGContextAddPath(context, leftCircles);
+    CGContextAddPath(context, rightCircles);
+    CGContextSetRGBFillColor(context, 0, 0, 1, 1);
+    switch (fillMode) {
+        case kCGPathFill:
+            CGContextFillPath(context);
+            break;
+        case kCGPathFillStroke:
+            CGContextFillPath(context);
+            CGContextStrokePath(context);
+            break;
+        case kCGPathEOFill:
+            CGContextEOFillPath(context);
+            break;
+        case kCGPathEOFillStroke:
+            CGContextEOFillPath(context);
+            CGContextStrokePath(context);
```

Comment:
so funny story!
Fill clears the path, and stroke will not have a path to stroke.

The sheer joy of using this `fillMode` method is that you can just do...

```
CGContextDrawPath(context, fillMode);
```

it will do the right thing :)